ygot/proto_test.go

```diff
@@ -0,0 +1,71 @@
+package ygot
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/gnmi/errdiff"
+	"google.golang.org/protobuf/proto"
+
+	wpb "github.com/openconfig/ygot/proto/ywrapper"
+	epb "github.com/openconfig/ygot/ygot/testdata/exschemapath"
+)
+
+func TestPathsFromProto(t *testing.T) {
+	tests := []struct {
+		desc             string
+		inMsg            proto.Message
+		wantPaths        map[string]interface{}
+		wantErrSubstring string
+	}{{
+		desc: "simple proto with a single populated path",
+		inMsg: &epb.Interface{
+			Description: &wpb.StringValue{Value: "hello"},
+		},
+		wantPaths: map[string]interface{}{
+			"/interfaces/interface/config/description": "hello",
+		},
+	}, {
+		desc: "example message - supported fields",
+		inMsg: &epb.ExampleMessage{
+			Bo: &wpb.BoolValue{Value: true},
+			By: &wpb.BytesValue{Value: []byte{1, 2, 3, 4}},
+			// De is currently unsupported, needs parsing of decimal64 values.
```

Comment:
Why isn't `double` protobuf's parallel to float64? https://developers.google.com/protocol-buffers/docs/proto3#scalar

Reply:
or decimal64 for YANG?

Reply:
There are a lot of discussions about this one -- YANG decided to use decimal64, and since this format is intended to be able to serialise on the wire exactly, we decided to use something that was fixed precision. For developer ease-of-use in the Go bindings we decided to use a floating point value.